minidump-processor/src/op_analysis.rs

```diff
@@ -183,6 +193,71 @@ mod amd64 {
         ret
     }
 
+    impl MemoryOperation {
+        fn from_amd64_instruction(i: Instruction) -> Self {
+            // TODO: Extend to other opcodes
+            match i.opcode() {
+                Opcode::LEA => Self::NoOperation,
+                Opcode::MOV
+                | Opcode::MOVAPD
+                | Opcode::MOVAPS
+                | Opcode::MOVBE
+                | Opcode::MOVD
+                | Opcode::MOVQ
+                | Opcode::MOVDDUP
+                | Opcode::MOVDQ2Q
+                | Opcode::MOVDQA
+                | Opcode::MOVDQU
+                | Opcode::MOVHLPS
+                | Opcode::MOVHPD
+                | Opcode::MOVHPS
+                | Opcode::MOVLHPS
+                | Opcode::MOVLPD
+                | Opcode::MOVLPS
+                | Opcode::MOVMSKPD
+                | Opcode::MOVMSKPS
+                | Opcode::MOVNTDQ
+                | Opcode::MOVNTI
+                | Opcode::MOVNTPD
+                | Opcode::MOVNTPS
+                | Opcode::MOVNTQ
+                | Opcode::MOVQ2DQ
+                | Opcode::MOVS
+                | Opcode::MOVSD
+                | Opcode::MOVSHDUP
+                | Opcode::MOVSLDUP
+                | Opcode::MOVSS
+                | Opcode::MOVSX
+                | Opcode::MOVSXD
+                | Opcode::MOVUPD
+                | Opcode::MOVUPS
+                | Opcode::MOVZX => {
+                    if i.operand_count() != 2 {
+                        tracing::warn!("mov instruction had incorrect operand count");
+                        return Self::Undetermined;
+                    }
+                    match (i.operand(0).is_memory(), i.operand(1).is_memory()) {
+                        (true, true) => Self::UnknownOperation,
+                        (false, true) => Self::Read,
+                        (true, false) => Self::Write,
+                        (false, false) => Self::NoOperation,
+                    }
+                }
+                Opcode::CALL
+                | Opcode::CALLF
+                | Opcode::JMP
+                | Opcode::JMPF
+                | Opcode::JMPE
+                | Opcode::RETURN
+                | Opcode::RETF
+                | Opcode::IRET
+                | Opcode::IRETD
+                | Opcode::IRETQ => Self::Execute,
```

Comment:
This is wrong and should be fixed. Exec violation should be checked with a different way. These instructions only modify `rip`. If there is access violation exec, it will crash in the instruction after.